src/hooks/index.ts

```diff
@@ -971,3 +940,46 @@ export const useBalances = (wallet: YoroiWallet): YoroiAmounts => {
 
   return Utxos.toAmounts(utxos, primaryTokenId)
 }
+
+export const AUTH_METHOD_KEY = '/appSettings/authMethod'
+export const useAuthMethod = (storage: Storage, options?: UseQueryOptions<AuthMethod, Error>) => {
+  const query = useQuery({
+    suspense: true,
+    queryKey: ['useAuthMethod'],
+    queryFn: async () => {
+      const authMethod = parseAuthMethod(await storage.getItem(AUTH_METHOD_KEY))
+      if (isAuthMethod(authMethod)) return authMethod
+      return Promise.reject(new Error('useAuthMethod invalid data'))
+    },
+    ...options,
+  })
+
+  return query.data
+}
+
+const parseAuthMethod = (data: string | null) => {
```

Comment:
should return a value, and not throw

is there a fallback value? if not, undefined

Reply:
undefined will work here, but if we fallback to something it means that we are recovering from some state that should have be migrated, isn't ?